File: JS/cedrus/src/models/ClassificationLevelModel.ts
import { isString } from "@juniper-lib/util";
import { INamed, isINamed } from "./INamed";
import { ITypeStamped, isTypeStamped } from "./ITypeStamped";

export interface ClassificationLevelModel extends INamed, ITypeStamped<"classificationLevel"> {
    description: string;
}

export function isClassificationLevelModel(obj: unknown): obj is ClassificationLevelModel {
    return isTypeStamped("classificationLevel", obj)
        && isINamed(obj)
        && "description" in obj
        && isString(obj.description);
}

File: NETCore/Libs/Cedrus/Controllers/V1/Models/ClassificationLevelModel.cs
using Juniper.Cedrus.Entities;

namespace Juniper.Cedrus.Controllers.V1;

public class ClassificationLevelModel
{
    public string TypeStamp => "classificationLevel";
    public int? Id { get; }
    public string? Name { get; }
    public string? Description { get; }

    public ClassificationLevelModel(ClassificationLevel? level)
    {
        Id = level?.Id;
        Name = level?.Name;
        Description = level?.Description;
    }
}

